### ISSUE TITLE
Weird bogofile errors in the Emacs binding

### ISSUE THREAD

**Issue by ane** [phildawes/racer#195]
Using Rust nightly 2015-04-12 on Ubuntu 14.04 and HEAD of racer, upon completing `use std::io::B` (the only contents of the file) from Emacs it doesn't work. On the command line it works fine. Tested with a clean emacs environment in only rust-mode and company enabled.

When I debug `racer--candidates` [(racer.el:56)] (https://github.com/phildawes/racer/blob/master/editors/racer.el#L56), while evaluating the let form of `lines` I get the following message from racer:

```
Result: ("PREFIX 13,14,B" "bogofile:1:15: 1:16 error: expected one of `::`, `;`, or `as`, found `<eof>`" 
"bogofile:1 use std::io::B" "                         ^" "bogofile:1:15: 1:16 error: expected one of `::`, `;`, or `as`, 
found `<eof>`" "bogofile:1 use std::io::B" "                         ^")
```

Newlines added to clarify the message. But if I invoke it on the command line it prints out a few lines of what apparently are correct results.

**Comment by phildawes** [comment]
Hi @ane,
I haven't been able to reproduce this on my setup. I created a clean user on ubuntu 14.04. My emacs and rust nightly versions are: 
```
$ emacs --version
GNU Emacs 24.3.1
$ rustc --version
rustc 1.0.0-nightly (6436e348e 2015-04-08) (built 2015-04-09)
```
I'm on rust-mode HEAD and racer HEAD.
I eval'd the following from .emacs file:
```
;; rust mode
(add-to-list 'load-path "/usr/local/src/rust-mode/")
(autoload 'rust-mode "rust-mode" nil t)
(add-to-list 'auto-mode-alist '("\\.rs\\'" . rust-mode))

;; racer
(setq racer-cmd "/home/clean/racer/target/release/racer")
(setq racer-rust-src-path "/usr/local/src/rust/src/")
(add-to-list 'load-path "/home/clean/racer/editors")
(eval-after-load "rust-mode" '(require 'racer))

;; company
(add-to-list 'load-path "/home/clean/company-0.8.2/")
(autoload 'company-mode "company" nil t)
```
Pressing tab after
```use std::io::B``` 
works ok. I will try pulling a more recent nightly tonight after work, but if things are working on the command line then I suspect that won't be it. Do my versions of the various pieces of software differ substantually?

Thanks,

Phil

**Comment by ane** [comment]
A-ha! I found what caused it! 

I had rust installed in `~/rust/`, so my `racer-rust-src-path` was set to that. On the command line, the shell racer runs in does the expansion for it, but I suspect that Emacs' `(setenv "RUST_SRC_PATH" racer-rust-src-path)` does *not* expand the `~` in it. 

I managed to fix it by adding `(expand-file-name racer-rust-src-path)) to the `setenv` calls. I can submit a PR for this later.
